### PR TITLE
Issue 6658: LTS - Enable using SlowChunkStorage with SlowStorageFactory

### LINKED ISSUE #6658: LTS - Enable using SlowChunkStorage with SlowStorageFactory
**Is your feature request related to a problem? Please describe.**
Currently SlowStorageFactory can create SlowStorage. 
Additionally we need it to  create ChunkedSegmentStorage with SlowChunkStorage 

**Describe the solution you'd like**
Based on config either inject SlowStorage or SlowChunkStorage 

**Describe alternatives you've considered**
Creating another storage factory just for this purpose is not a good option.

**Additional context**

### PR DESCRIPTION
**Pull request by sachin-j-joshi**
Signed-off-by: Sachin Joshi <sachin.joshi@emc.com>

**Change log description**  
Issue 6658: LTS - Enable using SlowChunkStorage with SlowStorageFactory

**Purpose of the change**  
Fixes #6658 

**What the code does**  
Based on config either inject `SlowStorage `or just `SlowChunkStorage `into the `ChunkedSegmentStorage`

- New config to control whether to wrap the entire operation or only the `ChunkStorage `operation.
- Small change to SimpleStorageFactory interface so that we can just create `ChunkStorage`
- Update existing factories and tests

**How to verify it**  
All tests should pass.

### CODE REVIEW

**Comment by RaulGracia** [comment]
@sachin-j-joshi is this still draft? Are you planning go ahead with it or close it?

**Comment by sachin-j-joshi** [comment]
@RaulGracia I'll take it out of draft mode after successful system tests.

**Comment by sachin-j-joshi** [comment]
Other system tests succeeded `test-pravega-continuous/4463/`

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/mocks/SlowStorageFactory.java [hunk 1]
```diff
@@ -51,8 +53,17 @@ public class SlowStorageFactory implements SimpleStorageFactory {
     @Override
     public Storage createStorageAdapter(int containerId, ChunkMetadataStore metadataStore) {
         if (inner instanceof SimpleStorageFactory) {
-            val innerStorage = ((SimpleStorageFactory) inner).createStorageAdapter(containerId, metadataStore);
-            return new SlowStorage(innerStorage, executor, config);
+            if (config.isSlowModeInjectChunkStorageOnly()) {
```
Comment (abhinb) [inline segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/mocks/SlowStorageFactory.java:56]: is the config.isSlowModeInjectChunkStorageOnly redundant, as its checked in createChunkStorage as well?

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/StorageExtraConfig.java [hunk 2]
```diff
@@ -33,6 +33,7 @@ public class StorageExtraConfig {
     public static final Property<Integer> STORAGE_WRITE_NO_OP_LATENCY = Property.named("noOp.write.latency.milliseconds", 20, "storageWriteNoOpLatencyMillis");
 
     public static final Property<Boolean> STORAGE_SLOW_MODE = Property.named("slow.enable", false);
+    public static final Property<Boolean> STORAGE_SLOW_MODE_INJECT_CHUNK_STORAGE_ONLY = Property.named("slow.inject.chunk.storage", true);
```
[the remark this answers is not in the document]
Comment (sachin-j-joshi) [reply segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/StorageExtraConfig.java:36]: yes.

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/mocks/SlowStorageFactory.java (hunk 1 above)
Comment (sachin-j-joshi) [reply segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/mocks/SlowStorageFactory.java:56]: SLTS has two sides
- Incoming calls from segment storage- `Storage` is the interface meant for the callers of SLTS i.e rest of the segment store.
- Outgoing calls to LTS - `ChunkStorage` is the interface individual LTS adapters implement.

In between those two end points there is 10K+ lines of SLTS code that converts call to Storage into appropriate interactions with LTS, manages metadata and deals with all sorts of failure scenarios etc.

So we could inject delays in two places
1. SlowChunkStorage  - add delays to actual calls going out to LTS - This delays only the single LTS calls.
2. SlowStorage - adds delays to all incoming Storage calls. This delays entire storage call.

So, No - config.isSlowModeInjectChunkStorageOnly is not redundant. It is used to decide whether we narrowly slow down only the actual LTS calls or slow down entire storage call.

**Comment by sachin-j-joshi** [comment]
@RaulGracia @abhinb do this change look good?

**Review by abhinb — APPROVED** [review]

**Review by RaulGracia — APPROVED** [review]

**Pull request merged**